### PR TITLE
Added support for the use container option when running lamdas locally

### PR DESCRIPTION
**Pull request by gsweetwood**
## Description

This allows users to locally run / debug lambda functions that have native package references.

## Motivation and Context

If the user has a lambda app that has native package references, they can now build and run inside a container locally.  This will also allow local run/debug for runtimes that are not installed locally.

## Related Issue(s)

Issue #370 

## Testing

All current tests pass and webpack compiles with no errors.  

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the changelog using the script `npm run newChange`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/templates/templates.json [hunk 1]
```diff
@@ -43,6 +43,11 @@
                                             "type": "string",
                                             "default": null,
                                             "description": "If specified, adds a --docker-network parameter to the SAM CLI.  If not specified, no parameter will be sent."
+                                        },
+                                        "useContainer": {
+                                            "type": "boolean",
+                                            "default": false,
+                                            "description": "If specified, adds a --use-container parameter to the SAM CLI. If not specified, no parameter will be sent. If your functions depend on packages that have natively compiled dependencies use this flag to build your function inside an AWS Lambda-like Docker container."
```
Comment (hunterwerlla) [inline src/templates/templates.json]: nit: `AWS Lambda-like Docker container` should be `SAM Docker container`

File: src/shared/codelens/localLambdaRunner.ts [hunk 2]
```diff
@@ -449,11 +460,11 @@ export async function invokeLambdaFunction(
     }
 }
 
-const getConfig = async (params: {
+export async function getConfig(params: {
```
Comment (bryceitoc9) [inline src/shared/codelens/localLambdaRunner.ts]: I'm a little conflicted if this should be exported from this file; I'm thinking we should move this to a new file. The reasoning being, this config file was originally JUST for run/debug concerns, but now this has expanded its scope. Consider a `samUtils.ts` file?

File: src/templates/templates.json (hunk 1 above)
Comment (awschristou) [reply src/templates/templates.json]: Text is currently sourced from https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-build.html

File: src/shared/codelens/localLambdaRunner.ts (hunk 2 above)
Comment (awschristou) [reply src/shared/codelens/localLambdaRunner.ts]: Yes, and suggest to rename it to something more specific, like "getSamLocalDebugConfiguration"

Comment (bryceitoc9) [reply src/shared/codelens/localLambdaRunner.ts]: Also move any functions that consume from this config to that file as well and export them.

**Comment by hunterwerlla** [comment]
> Looks good.
> 
> @hunterwerlla what do you think about an integration test that performs a local debug against a single runtime (python 3.8 for example) that is configured to build in containers?

That would work but I'm not sure we need to have an integration test, just a "is set, yep adds setting" test

File: src/templates/templates.json (hunk 1 above)
Comment (hunterwerlla) [reply src/templates/templates.json]: I disagree, the text is weird and should not be copied

**Pull request merged**